pkg/agent/validations.go

```diff
@@ -0,0 +1,86 @@
+	"sufficient-network-latency-requirement-for-role": "Network latency",
+	"sufficient-packet-loss-requirement-for-role":     "Packet loss",
+	"has-inventory":                                   "Hardware information",
+	"has-min-cpu-cores":                               "Minimum CPU cores",
+	"has-min-memory":                                  "Minimum Memory",
+	"has-min-valid-disks":                             "Minimum disks of required size",
+	"has-cpu-cores-for-role":                          "Minimum CPU cores for selected role",
+	"has-memory-for-role":                             "Minimum memory for selected role",
+	"hostname-unique":                                 "Unique hostname",
+	"hostname-valid":                                  "Valid hostname",
+	"connected":                                       "Connected",
+	"media-connected":                                 "Media Connected",
+	"machine-cidr-defined":                            "Machine CIDR",
+	"belongs-to-machine-cidr":                         "Belongs to machine CIDR",
+	"ignition-downloadable":                           "Ignition file downloadable",
+	"belongs-to-majority-group":                       "Belongs to majority connected group",
+	"valid-platform-network-settings":                 "Platform network settings",
+	"ntp-synced":                                      "NTP synchronization",
+	"container-images-available":                      "Container images availability",
+	"lso-requirements-satisfied":                      "LSO requirements",
+	"ocs-requirements-satisfied":                      "OCS requirements",
+	"sufficient-installation-disk-speed":              "Installation disk speed",
+	"cnv-requirements-satisfied":                      "CNV requirements",
+	"api-domain-name-resolved-correctly":              "API domain name resolution",
+	"api-int-domain-name-resolved-correctly":          "API internal domain name resolution",
+	"apps-domain-name-resolved-correctly":             "Application ingress domain name resolution",
+	"dns-wildcard-not-configured":                     "DNS wildcard not configured",
+	"non-overlapping-subnets":                         "Non overlapping subnets",
+	"vsphere-disk-uuid-enabled":                       "Vsphere disk uuidenabled",
+}
+
+func checkHostsValidations(cluster *models.Cluster, log *logrus.Logger) bool {
+
+	hostsValidationsOk := true
+
+	for _, h := range cluster.Hosts {
+		validationsInfo := common.ValidationsStatus{}
+		err := json.Unmarshal([]byte(h.ValidationsInfo), &validationsInfo)
+		if err != nil {
+			log.WithError(err).Error("Unable to verify cluster hosts validations")
+			return false
+		}
+
+		for category, validationResults := range validationsInfo {
+			for _, r := range validationResults {
+				switch r.Status {
+				case validationFailure, validationError:
+					hostsValidationsOk = false
+
+					label := r.ID
+					if v, ok := hostValidationLabels[r.ID]; ok {
+						label = v
+					}
```

Comment:
if we get a validation reference that we don't have in our map, should we log it?

Reply:
In that case the subsequent trace will log the result ID as received